packages/wdio-allure-reporter/src/@types/allure-js-commons.d.ts

```diff
@@ -0,0 +1,122 @@
+declare module 'allure-js-commons' {
+    declare class Allure {
+        constructor();
+
+        setOptions(options: Allure.Options): void;
+
+        getCurrentSuite(): Allure.Suite;
+
+        getCurrentTest(): Allure.Test;
+
+        startSuite(suiteName: string, timestamp?: number): void;
+
+        endSuite(timestamp?: number): void;
+
+        startCase(testName: string, timestamp?: number): void;
+
+        endCase(status: Allure.Status, err?: {}, timestamp?: number): void;
+
+        startStep(stepName: string, timestamp?: number): void;
+
+        endStep(status: Allure.Status, timestamp?: number): void;
+
+        setDescription(description: string, timestamp?: number): void;
+
+        addAttachment(attachmentName: string, buffer: any, type?: string): void;
+
+        pendingCase(testName: string, timestamp?: number): void;
+    }
+
+    export = Allure;
+    export as namespace Allure;
+    declare namespace Allure {
+        interface Options {
+            targetDir: string;
+        }
+
+        // type Status = "passed" | "pending" | "skipped" | "failed" | "broken";
+        type Status = "passed" | "pending" | "skipped" | "failed" | "broken" | "canceled";
+
+        enum TYPE {
+            TEXT = "text",
+            HTML = "html",
+            MARKDOWN = "markdown"
+        }
+
+        class Suite {
+            currentStep: Allure.Step;
+            testcases: string[];
+
+            constructor(name: string, timestamp?: number);
+
+            name: string;
+
+            end(timestamp?: number): void;
+
+            hasTests(): boolean;
+
+            addTest(test: Test): boolean;
+
+            toXML(): string;
+        }
+
+        class Test {
+            steps: Allure.Step[];
+
+            constructor(public name: string, timestamp?: number);
+
+            setDescription(description: string, type: TYPE): void;
+
+            addLabel(name: string, value: string): void;
+
+            addParameter(kind: any, name: string, value: string): void;
+
+            addStep(step: Step): void;
+
+            addAttachment(attachment: Attachment): void;
+
+            end(status: Status, error: Error, timestamp?: number): void;
+
+            toXML(): string;
+        }
+
+        class Description {
+            constructor(value: string, type: TYPE);
+
+            toXML(): string;
+        }
+
+        class Step {
+            attachments: Attachment[];
+
+            constructor(name: string, timestamp?: number);
+
+            addStep(step: Step): void;
+
+            addAttachment(attachment: Attachment): void;
+
+            end(status: Status, error: Error, timestamp?: number): void;
+
+            toXML(): string;
+        }
+
+        class Attachment {
+            constructor(title: string, source: any, size: number, mime: string);
+
+            addStep(step: Step): void;
+
+            addAttachment(attachment: Attachment): void;
+
+            end(status: Status, error: Error, timestamp?: number): void;
+

```

Comment:
can we clean up the empty lines between function definition within the classes? It doesn't follow other TS declarations and bloats up the file a bit

Reply:
This is the original file from https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/allure-js-commons
I added the reason not to use the types package on top. I wonder if it's worth it to refactor all those spaces.
```
// Allure is providing its own types in v2, but there's too many changes,
// so this is temporarily copied here to customize typings with allure 1
```

Reply:
If it is maintained in definitely types can't we just import it?

Reply:
@christian-bromann allure1 is obsolete and isn't maintained. So I see reasonable to copy `d.ts` if it makes easier to move wdio-allure to ts.

Reply:
The types in Definitely Types are not ideal, I can't use it without customization.